### ISSUE TITLE
Setting filters on the client is not working

### ISSUE THREAD

**Issue by cherbst** [alethes/meteor-pages#101]
Setting filters on the client does not work at all. Not even in the provided Example:
https://github.com/alethes/meteor-pages/tree/master/examples/iron-router

**Comment by doubletaketech** [comment]
What proper syntax for adding a filter?  I'm trying to filter a collection by a Session.  The code below returns page numbers in pageNav, but no rows of data.
```
if Meteor.isClient
  @MeterPages = new Meteor.Pagination Meters,
        templateName: "meters"
        dataMargin: 5
        fastRender: true
        perPage: 20
        filters:
          projectId:
                Session.get('selectedProjectId')
        table:
          class: "table"
          fields: fields
          header: _.map fields, (f) -> f[0].toUpperCase() + f.slice 1 #Capitalize fields

```

**Comment by evenfrost** [comment]
+1 @doubletaketech. How can a Session be used to filter data? Also, I'm trying to use some complex logic for filters, is there any possibility to provide function for pages data to be filtered by its return value (on the client, to use with Session)?

**Comment by doubletaketech** [comment]
I use sessions all the time to filter data based on something selected in the UI.  My assumption is that this package's required coffeescript is available on both client and server. My projectId is a Session set on the client. This session changes frequently and I need the paginated data to react and filter by projectId.

I too also have a need for more complex functions in the filters.
I've had to move on to writing my own pagination so I can write my own filter function.

**Comment by johntday** [comment]
using version 1.8.4
whenever i change filters on client side i get the following error:
"Changing filters not allowed"

**Comment by johntday** [comment]
using version 1.8.4
whenever i change sort on client side i get the following error:
"Changing sort not allowed"

**Comment by alethes** [comment]
This should be fixed since v1.8.0. Remember to define a policy for setting changes with `availableSettings` (eg. `availableSettings: {filters: true}` allows the client to change the filters under all circumstances).

---
As for the reactive changes in the settings, calling `Session.get(...)` on initialization will only pass the current value to the constructor. A simple solution (client-side):
```
Tracker.autorun(function(){ //or a template helper, or some other reactive computation
  MyPaginationObject.set({
    filters: Session.get("myFilters"),
    sort: Session.get("mySort")
  });
});
```
If you want to make eg. *filters* and *sort* settings available to all clients, remember to pass
`availableSettings: {filters: true, sort: true}` to the constructor.

**Comment by JDiPierro** [comment]
I'm having this issue in v1.8.4

    $ meteor list | grep pages
    alethes:pages               1.8.4  State of the art, out of the box Meteor pa...

My source:

    Images = new FS.Collection("images", {
        stores: [
            original_image_store,
            thumbnail_store
        ],
        filter: {
            allow: {
                contentTypes: ['image/*', 'video/webm']
            }
        }

    Wall_Pages = new Meteor.Pagination(Images, {
        debug: true,
        availableSettings: {
            filters: true,
            sort: true,
            limit: true
        },
        filters: {
            owner: Meteor.userId()
        },
        infinite: true,
        infiniteTrigger:.9,
        infiniteRateLimit: 1,
        infiniteStep: 1,
        dataMargin: 5,
        itemTemplate: "image_object",
        pageSizeLimit: 100,
        perPage: 10,
        maxSubscriptions: 500,
        router: "true",
        templateName: "wall",
        homeRoute: "/wall/:_id",
        routerTemplate: "app",
        sort: {
            id: 1
        }
    });

    Router.route('/wall/:_id', function () {
        var wall_id = this.params._id,
            wall = UserWalls.findOne(wall_id);
        Session.set('active_wall', wall_id);
        
        var attached_ids = [];
        if(wall.elements && wall.elements.length > 0) {
            attached_ids = _.map(wall.elements, function(value) {
                return value.id;
            });
        }
        Wall_Pages.set({filters: {
            _id: { $in: attached_ids}
        }});

        this.render('app_container');
    });

When I open a page to `localhost:3000/walls/:_id` I get the loading spinner and in console I see `Changing filters not allowed.`

Do I have something configured incorrectly?
